fix: added a dedicated test to confirm that serde_json correctly handles JSON strings with leading and trailing whitespace

tomasandroil opened the pull request:
added a dedicated test (`serde_json_accepts_whitespace_wrapped_json`) to confirm that `serde_json` correctly handles JSON strings with leading and trailing whitespace.

mattsse commented on crates/rpc/ipc/src/stream_codec.rs:
```diff
@@ -115,7 +115,8 @@ impl tokio_util::codec::Decoder for StreamCodec {
                     let bts = buf.split_to(idx + 1);
                     return match String::from_utf8(bts.into()) {
-                        Ok(val) => Ok(Some(val)),
+                        // Trim to remove surrounding whitespace from a valid JSON block
+                        Ok(val) => Ok(Some(val.trim().to_string())),
```
this now does another allocation, I don't think this is worth doing because serde_json deserialze can handle this case I believe, but I would accept a test for this when this returns leading whitespace

tomasandroil commented:
@mattsse 
I've removed the `.trim()` logic from the decoder as suggested and added a dedicated test (`serde_json_accepts_whitespace_wrapped_json`) to confirm that `serde_json` correctly handles JSON strings with leading and trailing whitespace.

mattsse commented on crates/rpc/ipc/src/stream_codec.rs:
```diff
@@ -115,7 +115,7 @@ impl tokio_util::codec::Decoder for StreamCodec {
                     let bts = buf.split_to(idx + 1);
                     return match String::from_utf8(bts.into()) {
-                        Ok(val) => Ok(Some(val)),
+                        Ok(val) => Ok(Some(val.to_string())),
```
```suggestion
                        Ok(val) => Ok(Some(val)),
```

mattsse commented on crates/rpc/ipc/src/stream_codec.rs:
```diff
@@ -210,14 +210,14 @@ mod tests {
-        // TODO: maybe actually trim it out
-        assert_eq!(request2, "\n\n\n\n{ test: 2 }");
+        // Leading/trailing whitespace is trimmed during decoding
+        assert_eq!(request2, "{ test: 2 }");
```
I think we need to undo this as well?

mattsse reviewed (changes requested):
ty, smol nit

tomasandroil commented:
@mattsse 
I've reverted the changes as requested
